database/models/fans.js

```diff
@@ -0,0 +1,132 @@
+            )
+          )
+          FROM event_fan ef, events e
+          WHERE ef.events_id = e.id
+          AND ef.fans_id = '${req.params.fanId}'
+          GROUP BY ef.fans_id
+        ),
+        'artists', (SELECT
+          array_to_json(
+            array_agg(
+              json_build_object(
+                'artist_id', a.id,
+                'artist_display-name', a.display-name,
+                'artist_instrument', a.instrument,
+                'artist_genre', a.genre,
+                'artist_bio', a.bio,
+                'artist_picture', a.picture,
+                'artist_paymentmethod', a.paymentmethod,
+                'artist_fans_count', a.fans_count
+              )
+            )
+          )
+          FROM artist_fan af, artist a
+          WHERE af.artists_id = a.id
+          AND af.fans_id = '${req.params.fanId}'
+          GROUP BY af.fans_id
+        )
+      )
+      FROM fans f
+      WHERE f.id = '${req.params.fanId}'
+    `)
+    .then((data) => {
+      res.statusCode = 200;
+      res.end(JSON.stringify(data));
+    })
+    .catch((err) => {
+      res.status(500);
+      console.log('GET Fan Info error: ', err);
+      res.end(JSON.stringify(err));
+    });
+  },
+  saveEvent: function(req, res) {
+    client.query(`INSERT INTO event_fan
+                  (fans_id, events_id)
+                  VALUES
+                  (${req.body.id}, ${req.body.event_id})`)
+    .then(() => res.sendStatus(201);)
+    .catch(err => {
+      console.log('Save Event Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  followArtist: function(req, res) {
+    client.query(`INSERT INTO artist_fan
+                  (fans_id, artists_id)
+                  VALUES
+                  (${req.body.id}, ${req.body.artist_id})`)
+    .then(() => res.sendStatus(201);)
+    .catch(err => {
+      console.log('Follow Artist Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  put: function(req, res) {
+    client.query(`UPDATE fans
+                  SET street = ${req.body.street},
+                      city = ${req.body.city},
+                      state = ${req.body.state}
+                  WHERE id = ${req.params.fanId}`)
+    .then(() => res.sendStatus(204);)
+    .catch(err => {
+      console.log('Update profile Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  removeEvent: function(req, res) {
+    client.query(`DELETE FROM event_fan
+                  WHERE fans_id = ${req.params.fanId}
+                  AND events_id = ${req.params.eventId} `)
+    .then(() => res.sendStatus(201);)
+    .catch(err => {
+      console.log('Remove event Fan error: ', err);
+      res.status(500);
+      res.end(JSON.stringify(err));
+    });
+  },
+  unfollowArtist: function(req, res) {
+    client.query(`DELETE FROM artist_fan
+                  WHERE fans_id = ${req.params.fanId}
```

Comment:
artist_fan table is named as 'art_fan' in database / (fans_id, artists_id) are named as (art_id, fan_id) in art_fan table